libs/algo/odc/algo/io.py

```diff
@@ -84,17 +86,30 @@ def _load_with_native_transform_1(
 
     (ds,) = sources.data[0]
     load_geobox = compute_native_load_geobox(geobox, ds, basis)
+
     if pad is not None:
         load_geobox = gbox.pad(load_geobox, pad)
 
     mm = ds.type.lookup_measurements(bands)
     xx = Datacube.load_data(sources, load_geobox, mm, dask_chunks=load_chunks)
-    xx = native_transform(xx)
 
+
+    xxs = native_transform(xx)
+    xxss = []
     if groupby is not None:
         if fuser is None:
             fuser = _nodata_fuser  # type: ignore
-        xx = xx.groupby(groupby).map(fuser)
+        if isinstance(xxs, list):
+            idx = 0
+            for x in xxs:
+                xxss.append(x.groupby(groupby).map(fuser, [idx]))
+                idx += 1
+
+            xx = xxss[0]
+            for xs in xxss[1:]:
+                xx =  np.logical_or(xx, xs)
+        else:
+           xx = xxs.groupby(groupby).map(fuser)
```

Comment:
I don't agree the mask bands should be merged here. Especially when `fuser` is involved. It could cause the same issue that I observed in wofs plugin. When the datasets are in the same crs, they're merged in native crs, when not, they're merged in reprojected crs. We need some consistency here, either before or after reprojection.